src/Repository/Project/PhaseRepository.php

```diff
@@ -24,4 +25,31 @@ public function getByEtudeQuery(Etude $etude)
 
         return $qb;
     }
+
+    // Return the phases of a PV if set, or all the phases with no Proces Verbal if PV is null
+
+    /**
+     * @return \Doctrine\ORM\QueryBuilder
+     */
+    public function getByEtudeAndPv(Etude $etude, ProcesVerbal $pv = null)
+    {
+        if (isset($pv)) {
+            $qb = $this->_em->createQueryBuilder()
+            ->select('p')
+            ->from(Phase::class, 'p')
+            ->where('p.etude = :etude')
+            ->andwhere('p.procesVerbal IS NULL OR p.procesVerbal = :pv_id')
+            ->setParameter('etude', $etude)
+            ->setParameter('pv_id', $pv->getId());
+        } else {
+            $qb = $this->_em->createQueryBuilder()
+            ->select('p')
+            ->from(Phase::class, 'p')
+            ->where('p.etude = :etude')
+            ->andwhere('p.procesVerbal IS NULL')
+            ->setParameter('etude', $etude);
+        }
+
+        return $qb;
+    }
```

Comment:
Pour accélerer la review, voici le code optimal pour cette fonction. Testé en local

```suggestion
        /**
     * Used to display phase selector in ProcesVerbalType.
     * If a PV is set, return a query builder to all phases not associated to a PV & the ones associated with given PV
     * If no PV is set, returna query builder to all phases not associated to a PV
     *
     * @return \Doctrine\ORM\QueryBuilder
     */
    public function getByEtudeAndPv(Etude $etude, ProcesVerbal $pv = null)
    {
        $qb = $this->_em->createQueryBuilder()
            ->select('p')
            ->from(Phase::class, 'p')
            ->where('p.procesVerbal IS NULL');

        if ($pv !== null) {
            $qb->where('p.procesVerbal IS NULL OR p.procesVerbal = :pv_id')
            ->setParameter('pv_id', $pv->getId());
        }

        $qb->andWhere('p.etude = :etude')
            ->setParameter('etude', $etude);

        return $qb;
    }
```